tests/taproot_maker_abort2.rs

```diff
@@ -0,0 +1,210 @@
+#[test]
+fn test_taproot_maker_abort2() {
+    // ---- Setup ----
+    warn!("🧪 Running Test: Taproot Maker Abort 2");
+
+    // Create one normal maker, and a maker that closes connection at PrivateKeyHandover step
+    let makers_config_map = vec![
+        (7103, Some(19071), MakerBehavior::Normal),
+        (7104, Some(19072), MakerBehavior::CloseAtPrivateKeyHandover),
+    ];
+    let taker_behavior = vec![TakerBehavior::Normal];
+
+    // Initialize test framework
+    let (test_framework, mut taproot_taker, taproot_makers, block_generation_handle) =
+        TestFramework::init_taproot(makers_config_map, taker_behavior);
+
+    let bitcoind = &test_framework.bitcoind;
+    let taproot_taker = taproot_taker.get_mut(0).unwrap();
+
+    info!("💰 Funding taker and maker");
+
+    // Fund the Taproot Taker with 3 UTXOs of 0.05 BTC each
+    let taproot_taker_original_balance =
+        fund_taproot_taker(taproot_taker, bitcoind, 3, Amount::from_btc(0.05).unwrap());
+
+    // Fund the Taproot Maker with 4 UTXOs of 0.05 BTC each
+    fund_taproot_makers(
+        &taproot_makers,
+        bitcoind,
+        4,
+        Amount::from_btc(0.05).unwrap(),
+    );
+
+    // Start the Taproot Maker Server thread
+    info!("🚀 Initiating Maker server...");
+
+    let taproot_maker_threads = taproot_makers
+        .iter()
+        .map(|maker| {
+            let maker_clone = maker.clone();
+            thread::spawn(move || {
+                start_maker_server_taproot(maker_clone).unwrap();
+            })
+        })
+        .collect::<Vec<_>>();
+
+    // Wait for taproot maker to complete setup
+    for maker in &taproot_makers {
+        while !maker.is_setup_complete.load(Relaxed) {
+            info!("⏳ Waiting for taproot maker setup completion");
+            thread::sleep(Duration::from_secs(10));
+        }
+    }
+
+    // Sync wallets after setup
+    for maker in &taproot_makers {
+        maker.wallet().write().unwrap().sync_and_save().unwrap();
+    }
+    let actual_maker_spendable_balances = verify_maker_pre_swap_balance_taproot(&taproot_makers);
+    info!("🔄 Initiating taproot maker abort 2 case. (Will fail mid-swap due to one maker closing connection at handling private key step)");
+
+    // Swap params - small amount for faster testing
+    let swap_params = SwapParams {
+        send_amount: Amount::from_sat(500000), // 0.005 BTC
+        maker_count: 2,
+        tx_count: 3,
+        required_confirms: 1,
+        manually_selected_outpoints: None,
+    };
+
+    // Attempt the swap - it will fail when maker closes connection
+    // After recovery, do_coinswap returns Ok(None) to indicate recovery was triggered
+    match taproot_taker.do_coinswap(swap_params) {
+        Ok(Some(_report)) => {
+            panic!("Swap should have failed due to maker closing connection, but succeeded with report!");
+        }
+        Ok(None) => {
+            info!("✅ Taproot coinswap triggered recovery as expected (Ok(None))");
```

Comment:
This test actually returns None, while we should consider it as a valid swap too. It just has script_path spend instead of key_path.

This can be fixed later. Noting it down so we can open an Issue and track. `Fix: Return swap report even if recovered via hashlock`.